sherpa/plot/__init__.py

```diff
@@ -113,11 +116,91 @@
            'Confidence1D', 'Confidence2D',
            'IntervalProjection', 'IntervalUncertainty',
            'RegionProjection', 'RegionUncertainty',
+           'TemporaryPlottingBackend',
+           'set_backend',
            )
 
 _stats_noerr = ('cash', 'cstat', 'leastsq', 'wstat')
 
 
+def set_backend(name):
+    '''Set the Sherpa plotting backend.
+
+    Plotting backends are registered in Sherpa with a string name.
+    See the examples below for how to get a list of available backends.
+
+    Parameters
+    ----------
+    name : string
+        Name of the plotting backend.
+
+    Example
+    -------
+    Set the backend to use Pylab from matplotlib for plotting. This is
+    probably what most users need:
+
+        >>> from sherpa.plot import set_backend
+        >>> set_backend('pylab')
+
+    Get a list of registered backends:
+
+        >>> from sherpa.plot import PLOT_BACKENDS
+        >>> PLOT_BACKENDS
+        {'BaseBackend': sherpa.plot.backends.BaseBackend,
+         'BasicBackend': sherpa.plot.backends.BasicBackend,
+         'IndepOnlyBackend': sherpa.plot.backends.IndepOnlyBackend,
+         'pylab': sherpa.plot.pylab_backend.PylabBackend}
+
+    This list shows the names and the class for each backend.
+    Details for each backend can be found in the Sherpa documentation or by
+    inspecting the backend classes using normal Python functionality:
+
+        >>> from sherpa.plot.backends import BasicBackend
+        >>> help(BasicBackend)
+    '''
+    global backend
+    if name in PLOT_BACKENDS:
+        backend = PLOT_BACKENDS[name]()
+    else:
+        warning(f'Plotting backend not changed because {name} is not a known plotting backend.\n' +
+                f'List of available plotting backends: {list(PLOT_BACKENDS.keys())}',
+                )
+
```

Comment:
I am wondering if we want something like

```python
def list_backends() -> List[str]:
    """An inspiring docstring"""

    return sorted(PLOT_BACKENDS)
```

The idea being that this hides the `PLOT_BACKENDS` dictionary from the user, and so if we ever get to entrypoints/whatever a user may not have to change their code even if we replace the lower-level code.

Reply:
I thought about that a `list_backends` function and rejected the idea, but I'm not feeling very strongly and could be convinced otherwise. Here are the reasons why I decided against it:

- Most users won't change backends at all. They'll just use the pylab that's set as default.
- Those that do are unlikely to use any of the backends except pylab and maybe bokeh. They'll install (or already have) the dependencies and do exactly what the narrative docs or CIAO threads will say. say: Type `set_backend('bokeh')` and then press enter. They don't need that function either.
- Sherpa has a large number of functions (in particular in the ui) that just execute one or two lines of code, e.g. "list_stats". However, there is a cost to it - it needs to be documented, tested, and explained in narrative docs. Users need to learn those methods as opposed to using Python structure they already know - and by now I think many more people do know Python than when Sherpa was originally conceived. So, for every new method, I try to ask myself: Do we really need that or is standard Python just fine? Here, I firmly believe that it's just fine. 
- In this error message, the `PLOT_BACKENDS` dictionary is already hidden from the user, for the reasons you mention. This is an f-string, so the user will see `List of available plotting backends: BasicBackend, pylab, IndepOnlyBackend`.

Reply:
As a side note on

> Users need to learn those methods as opposed to using Python structure they already know

This is true, but it also exposes the underlying data representation to the user. So it's a balance, and it depends on the command in question/structures in use.

The question about `list_backend` not a strong suggestion, so doesn't need to be acted on. I will note that I dislike the general design pattern of "oh, to get at this information (e.g. list of arguments), call this function with an invalid argument and see what the response is" (but again it's a question of balance, and I'm not arguing for this function here).

Reply:
I've looked into generating the docstring dynamically, which I think would be ideal here but it's not really possible to do that. How about we add the build-in formats hardcoded into to the doctring? That's going to catch 99% of the use cases - and those people who write their own plotting backends will know what they do, because they need to look at the code anyway.